packages/forms/src/UIForm/fields/MultiSelectTag/MultiSelectTag.component.js

```diff
@@ -0,0 +1,241 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import keycode from 'keycode';
+import Typeahead from 'react-talend-components/lib/Typeahead';
+import Badge from 'react-talend-components/lib/Badge';
+import FieldTemplate from '../FieldTemplate';
+
+import theme from './MultiSelectTagWidget.scss';
+
+function escapeRegexCharacters(str) {
+	return str.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+}
+
+function getNewItemText(value) {
+	return `${value} (new)`;
+}
+
+function getLabel(titleMap, value) {
+	const itemConf = titleMap.find(item => item.value === value);
+	return itemConf ? itemConf.name : value;
+}
+
+export default class MultiSelectTag extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = { value: '' };
+		this.theme = {
+			container: theme.typeahead,
+			itemsContainer: theme['items-container'],
+			itemsList: theme.items,
+		};
+
+		this.onChange = this.onChange.bind(this);
+		this.onFocus = this.onFocus.bind(this);
+		this.onKeyDown = this.onKeyDown.bind(this);
+		this.onRemoveTag = this.onRemoveTag.bind(this);
+		this.onAddTag = this.onAddTag.bind(this);
+		this.resetSuggestions = this.resetSuggestions.bind(this);
+	}
+
+	/**
+	 * Manage suggestion selection
+	 * @param event
+	 * @param focusedItemIndex
+	 * @param newFocusedItemIndex
+	 */
+	onKeyDown(event, { focusedItemIndex, newFocusedItemIndex }) {
+		switch (event.which) {
+		case keycode.codes.enter:
+			if (!this.state.suggestions) {
+				break;
+			}
+			event.preventDefault();
```

Comment:
Shouldn't the `event.preventDefault` be done first?

Reply:
In fact the if statement should be removed. When you have focus on input, the suggestions are always truthy (even if the array is empty).
This was to allow user to submit on ENTER key but the suggestions are always opened and should the ENTER key should be about the typeahead.